stage1_fly/enter/main.go

```diff
@@ -47,10 +50,30 @@ func getRootDir(pid string) (string, error) {
 	return os.Readlink(rootLink)
 }
 
-func execArgs() error {
+// readEnv reads the environment from the env file written by stage1 run
+func readEnv() ([]string, error) {
+	var env []string
+	cwd, err := os.Getwd()
+	if err != nil {
+		return env, err
```

Comment:
Minor: when error-returning due to error condition, please prefer the idiom `return nil, err`.